bfd/elfnn-riscv.c

```diff
@@ -5142,7 +5227,31 @@ _bfd_riscv_relax_section (bfd *abfd, asection *sec,
 		 need the more rigorous checking for this optimization.  */
 	      undefined_weak = true;
 	    }
-
+          /* conditions to check ie -> le relaxation */
+          if (relax_func == _bfd_riscv_relax_tls_ie_to_le )
+	    {
+              bool need_reloc = false;
+              int ind=0;
+              bool dyn = htab->elf.dynamic_sections_created;
+              RISCV_TLS_GD_IE_NEED_DYN_RELOC(info, dyn, h, ind, need_reloc);
+              /* dont relax ie -> le if the relocation requires us to make dynamic relocations or when there are weak definitions or when we are not creating exec*/
```

Comment:
/* ToDo : don't relax for weak definitions  and any other visibility to be considered .*/